test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Smart/SmartSearchTests.cs

```diff
@@ -748,5 +752,278 @@ private void ConfigureFhirRequestContext(
+
+            var scopeRestriction = new ScopeRestriction("Patient", Core.Features.Security.DataActions.Read, "patient");
+
+            ConfigureFhirRequestContext(_contextAccessor, new List<ScopeRestriction>() { scopeRestriction });
+            _contextAccessor.RequestContext.AccessControlContext.CompartmentId = "smart-patient-A";
+            _contextAccessor.RequestContext.AccessControlContext.CompartmentResourceType = "Patient";
+
+            var updatedPatient = new Patient
+            {
+                Id = "smart-patient-A",
+                Name = new List<HumanName> { new HumanName().WithGiven("ShouldFail").AndFamily("Update") },
+            };
+
+            // Should fail when trying to update with only Read permission
+            await Assert.ThrowsAsync<UnauthorizedAccessException>(() => PutResource(updatedPatient));
+        }
+
+        [SkippableFact]
+        public async Task GivenSmartV2MultipleScopes_WhenUsingDifferentOperations_ThenOnlyAuthorizedOperationsSucceed()
+        {
+            Skip.If(
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4 &&
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4B,
+                "This test is only valid for R4 and R4B");
+
+            // Grant Search + Read permissions only (no Create or Update)
+            var scopeRestriction1 = new ScopeRestriction("Patient", Core.Features.Security.DataActions.Search, "patient");
+            var scopeRestriction2 = new ScopeRestriction("Patient", Core.Features.Security.DataActions.Read, "patient");
+
+            ConfigureFhirRequestContext(_contextAccessor, new List<ScopeRestriction>() { scopeRestriction1, scopeRestriction2 });
+            _contextAccessor.RequestContext.AccessControlContext.CompartmentId = "smart-patient-A";
+            _contextAccessor.RequestContext.AccessControlContext.CompartmentResourceType = "Patient";
+
+            // Search should succeed
+            var query = new List<Tuple<string, string>>();
+            query.Add(new Tuple<string, string>("_id", "smart-patient-A"));
+            var searchResults = await _searchService.Value.SearchAsync("Patient", query, CancellationToken.None);
+            Assert.NotEmpty(searchResults.Results);
+
+            // Read should succeed
+            var readResult = await _fixture.GetResourceHandler.Handle(
+                new GetResourceRequest(new ResourceKey("Patient", "smart-patient-A"), bundleResourceContext: null),
+                CancellationToken.None);
+            Assert.NotNull(readResult.Resource);
+
+            // Create should fail
+            var newPatient = new Patient
+            {
+                Id = "smart-v2-should-fail-create",
+                Name = new List<HumanName> { new HumanName().WithGiven("ShouldFail").AndFamily("Create") },
+            };
+            await Assert.ThrowsAsync<UnauthorizedAccessException>(() => PutResource(newPatient));
```

Comment:
Should we also be checking with PostResource?

Reply:
Updated.